lms/services/canvas.py

```diff
@@ -1,36 +1,79 @@
 from typing import Optional
 
 from lms import models
-from lms.services.exceptions import CanvasFileNotFoundInCourse
+from lms.services.exceptions import CanvasAPIPermissionError, CanvasFileNotFoundInCourse
 
 
 class CanvasService:
     """A high level Canvas service."""
 
     api = None
 
-    def __init__(self, canvas_api):
+    def __init__(self, canvas_api, file_service):
         self.api = canvas_api
-
-    def public_url_for_file(self, file_id, course_id, check_in_course=False):
-        """
-        Get a public URL for a Canvas file.
-
-        This will also attempt to check if the file is in the course to detect
-        course copy situations and fix it if we can.
-
-        :param file_id: The file to look up
-        :param course_id: The course the file should be in
-        :param check_in_course: Raise CanvasFileNotFoundInCourse if file_id isn't in
-            course_id
-        :return: A URL suitable for public presentation of the file
-        :raise  CanvasFileNotFoundInCourse: if check_in_course=True and file_id isn't in course_id
-            be in the course but isn't.
-        """
-
-        if check_in_course:
-            if not self.can_see_file_in_course(file_id, course_id):
-                raise CanvasFileNotFoundInCourse(file_id)
+        self._file_service = file_service
+
+    def public_url_for_file(
+        self, module_item_configuration, file_id, course_id, check_in_course=False
+    ):
```

Comment:
Oh, that's weird. I didn't mean to delete the docstring. I'll put it back

Reply:
https://github.com/hypothesis/lms/pull/2899/commits/b88be4a4aa8922a638695dcc56ebf6eb0d2cc980

I put the original docstring back and updated it (adding the `module_item_configuration` param) and also fixed it up a bit:

* Added that `CanvasAPIPermissionError` could be raised

* Removed `This will also attempt to check if the file is in the course to detect course copy situations and fix it if we can` from the docstring body because it's not true. On master there is no support for fixing course copy situations yet, so this docstring has been a lie on master since it was added. But also the `check_in_course` doesn't really detect course copy situations, it detects that the current user can't see the file in the course which could be for a number of reasons. Also it's not really true to say "This will also attempt to check if the file is in the course...": by default it won't do that, only if `check_in_course=True` is passed. I think it's better to leave the `:param check_in_course:` and `:raise CanvasFileNotFoundInCourse:` parts of the docstring to explain this, they cover it.

* Removed the `:return:`. If the first line of the docstring is `Return a public URL for file_id` then it doesn't add anything to also have `:return: A URL suitable for public presentation of the file` ("URL suitable for public presentation" is a great way to say "public URL" too :smile: )

* Fixed up some other gems: `:raise CanvasFileNotFoundInCourse: if check_in_course=True and file_id isn't in course_id be in the course but isn't` :)

* General wording tidy up